### PR TITLE
feat(bigquery):  Add BigQuery BigLake partitioned & un-partitioned examples

### PR DESCRIPTION
**Pull request by nevzheng**
## Description

Fixes #<ISSUE-NUMBER>

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [x] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [x] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [x] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [x] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [x] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):
+ https://cloud.google.com/bigquery/docs/create-cloud-storage-table-biglake#create-table-storage

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [x] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Comment by nevzheng** [comment]
this CL is dependent on the connection API which I added in go/tf-samples/pull/470, which should be merged prior to re-running the build.

File: bigquery/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (bharathkkb) [inline bigquery/bigquery_create_unpartitioned_biglake_table/main.tf:84]: This should be project id (or can be omitted all together)

Comment (nevzheng) [reply bigquery/bigquery_create_unpartitioned_biglake_table/main.tf:84]: done

**Comment by msampathkumar** [comment]
@nevzheng All the testing are passing! Nice work.

There are few minor suggestion for you.

File: bigquery/bigquery_create_partitioned_biglake_table/main.tf
Comment (msampathkumar) [inline bigquery/bigquery_create_partitioned_biglake_table/main.tf:39]: Two suggestion.
(1) In like 27, you have stated the purpose of before the `resource` definition. You can do the same here.
(2) Here you are using `/*` comments. Not a issues but I suggest simplifying your code comment. Prefer using CGC docs for detailed explanation. As much possible, keep your comment as simple as possible.

File: bigquery/bigquery_create_partitioned_biglake_table/main.tf
Comment (msampathkumar) [inline bigquery/bigquery_create_partitioned_biglake_table/main.tf:69]: use `#` instead of `##`. I suggest to use CGC docs for better explanations.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (adhiggs) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:77]:
```suggestion
# This creates a BigQuery table named "my_table" in the dataset "default".
# The table has a single column named "column1", which is of type STRING
# and is nullable.
```

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (adhiggs) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:81]:
```suggestion
   depends_on = [time_sleep.wait_7_min]
```

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (adhiggs) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:101]:
```suggestion
# This defines an external data configuration for the BigQuery table 
# that reads Parquet data from the publish directory of the default
# Google Cloud Storage bucket.
```

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (adhiggs) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:112]:
```suggestion
```

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (adhiggs) [inline bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:129]:
```suggestion
```

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (adhiggs) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:119]:
```suggestion
```

**Review by adhiggs — CHANGES_REQUESTED** [review]
Gave the comments a pass. Please verify accuracy of content.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:21]: done

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:25]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:27]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:46]: done

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:52]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:55]: done

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:62]: done

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:74]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:76]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:77]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:80]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:87]: done.

lgtm

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:91]: done

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:95]: done

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:108]: done

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:112]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:122]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:129]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:29]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:40]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:43]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:50]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:62]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:64]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:65]: done

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:68]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:76]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:79]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:83]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:103]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:109]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:114]: done.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:121]: done.

File: bigquery/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/bigquery_create_partitioned_biglake_table/main.tf]: ack. integrated adriaan's (tech writer) input

File: bigquery/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/bigquery_create_partitioned_biglake_table/main.tf]: ack. integrated adriaan's (tech writer) input

**Review by adhiggs — CHANGES_REQUESTED** [review]

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf]: done.

**Review by adhiggs — APPROVED** [review]
LGTM! Thank you.

**Review by msampathkumar — APPROVED** [review]
LGTM. All the tests are passing.

**Comment by msampathkumar** [comment]
Hi @nevzheng, I see the other approval but you would need approval from your [CODEOWNERS](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS) team which is terraform-google-modules/bigquery-terraform-swe.

Note: @terraform-google-modules/bigquery-terraform-swe is TeamSync managed.

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (wj-chen) [inline bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:89]: I see this was taken from https://cloud.google.com/bigquery/docs/create-cloud-storage-table-biglake#terraform, can we fix the indentation here (and there)?

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (wj-chen) [inline bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:96]: Align the indentation?

**Review by wj-chen — COMMENTED** [review]
Nicely done, thank you!

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (wj-chen) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:89]: Just fixing it once here then importing this example into the Terraform tab of that doc page should work.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:96]: ack.

this conforms to the terrafrom auto-formatter

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf:89]: done.

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf]: i applied `jsonencode` to get rid of those confusing EOFs

File: bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf
Comment (nevzheng) [reply bigquery/biglake/bigquery_create_unpartitioned_biglake_table/main.tf]: i applied `jsonencode` to get rid of those confusing EOFs

File: bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf
Comment (wj-chen) [inline bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:96]: Could you double check that this is working as intended? All the examples I saw had the TF syntax of using "=" instead of ":" in the input

Comment (nevzheng) [reply bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf:96]: A. Partitioned Table. I ran the up to date script in cloud shell. Resulting Schema in Pantheon
image

B. Unpartitioned Table. I ran the up to date script in cloud shell. Resulting Schema in Pantheon
image

**Review by wj-chen — APPROVED** [review]

**Pull request merged**
